src/Resolvers/Contracts/CachedTenantResolver.php

```diff
@@ -34,20 +34,25 @@ public function resolve(...$args): Tenant
             return $this->resolveWithoutCache(...$args);
         }
 
-        $key = $this->getCacheKey(...$args);
-
-        if ($tenant = $this->cache->get($key)) {
+        if ($tenant = $this->cache->get($this->getCacheKey(...$args))) {
             $this->resolved($tenant, ...$args);
 
             return $tenant;
         }
 
-        $tenant = $this->resolveWithoutCache(...$args);
-        $this->cache->put($key, $tenant, static::$cacheTTL);
+        $this->writeCache(...$args);
```

Comment:
How would the `return $tenant` work now?

Reply:
You are right 😊 Now `writeCache` returns the tenant. Maybe we could even rename it to `resolveAndCache` to make it clearer?